### PR TITLE
fixes improper syntax of Get-Member

### PR DESCRIPTION
**Pull request by jrquint**
Bug fix for win_updates:  cannot pipe variable to Get-Member, must use -InputObject

### CODE REVIEW

File: windows/win_updates.ps1 [hunk 1]
```diff
@@ -331,13 +331,13 @@ Function RunAsScheduledJob {
...
       # FUTURE: add a global WaitHandle in case another instance kills our job, so we don't block forever
-      $job = Wait-Job -Name $schedjob.Name -ErrorAction SilentlyContinue 
+      $job = Wait-Job -Name $schedjob.Name -ErrorAction SilentlyContinue
   }
 
   $sw = [System.Diagnostics.Stopwatch]::StartNew()
 
   # NB: output from scheduled jobs is delayed after completion (including the sub-objects after the primary Output object is available)
-  While (($job.Output -eq $null -or -not ($job.Output | Get-Member -Name Keys) -or -not $job.Output.Keys.Contains('job_output')) -and $sw.ElapsedMilliseconds -lt 15000) {
+  While (($job.Output -eq $null -or -not (Get-Member -InputObjet $job.Output -Name Keys) -or -not $job.Output.Keys.Contains('job_output')) -and $sw.ElapsedMilliseconds -lt 15000) {
```
Comment (kevinkenny) [inline windows/win_updates.ps1]: Surely `-InputObjet` is a typo? Should be `-InputObject` ?

Comment (jrquint) [inline windows/win_updates.ps1]: Ah, yes, that is a typo.  Looks like I 'backspaced' after I committed and and confirmed it working on my local repository.

##

**Comment by gregdek** [comment]
Thanks @jrquint. @mattdavispdx please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by nitzmahone** [comment]
Can you describe the original symptom you're trying to correct with this PR? Piping to Get-Member works fine for me:

```
PS C:\Users\vagrant> $bla = @{}
PS C:\Users\vagrant> $bla | Get-Member -Name Keys

   TypeName: System.Collections.Hashtable

Name MemberType Definition
---- ---------- ----------
Keys Property   System.Collections.ICollection Keys {get;}

PS C:\Users\vagrant> $bla | Get-Member -Name Bogus
PS C:\Users\vagrant>
```

**Comment by gregdek** [comment]
@jrquint A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by acaire** [comment]
I stumbled across this issue bootstrapping against a brand new AWS instance:

```
TASK [Windows Update] *******************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: +~~~~~~~~~~~~~~~~~~~~~
fatal: [that_instance]: FAILED! =>
{
    "changed": false,
    "error_record": {
        "CategoryInfo": {
            "Activity": "Get-Member",
            "Category": 2,
            "Reason": "InvalidOperationException",
            "TargetName": "",
            "TargetType": ""
        },
        "ErrorDetails": null,
        "Exception": {
            "Data": {},
            "HResult": -2146233079,
            "HelpLink": null,
            "InnerException": null,
            "Message": "You must specify an object for the Get-Member cmdlet.",
            "Source": null,
            "StackTrace": null,
            "TargetSite": null
        },
        "FullyQualifiedErrorId": "NoObjectInGetMember,Microsoft.PowerShell.Commands.GetMemberCommand",
        "InvocationInfo": {
            "BoundParameters": {},
            "CommandOrigin": 1,
            "DisplayScriptPosition": null,
            "ExpectingInput": false,
            "HistoryId": 1,
            "InvocationName": "Get-Member",
            "Line": "  While (($job.Output -eq $null -or -not ($job.Output | Get-Member -Name Keys) -or -not $job.Output.Keys.Contains('job_output')) -and $sw.ElapsedMilliseconds -lt 15000) {\n",
            "MyCommand": {
                "CommandType": 8,
                "DefaultParameterSet": null,
                "Definition": "\r\nGet-Member [[-Name] <string[]>] [-InputObject <psobject>] [-MemberType <PSMemberTypes>] [-View <PSMemberViewTypes>] [-Static] [-Force] [<CommonParameters>]\r\n",
                "HelpFile": "Microsoft.PowerShell.Commands.Utility.dll-Help.xml",
                "ImplementingType": "Microsoft.PowerShell.Commands.GetMemberCommand",
                "Module": "Microsoft.PowerShell.Utility",
                "ModuleName": "Microsoft.PowerShell.Utility",
                "Name": "Get-Member",
                "Noun": "Member",
                "Options": 1,
                "OutputType": "Microsoft.PowerShell.Commands.MemberDefinition",
                "PSSnapIn": null,
                "ParameterSets": "[[-Name] <string[]>] [-InputObject <psobject>] [-MemberType <PSMemberTypes>] [-View <PSMemberViewTypes>] [-Static] [-Force] [<CommonParameters>]",
                "Parameters": "System.Collections.Generic.Dictionary`2[System.String,System.Management.Automation.ParameterMetadata]",
                "RemotingCapability": 0,
                "Verb": "Get",
                "Visibility": 0
            },
            "OffsetInLine": 57,
            "PSCommandPath": "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ansible-tmp-1465288618.21-196443209232539\\win_updates.ps1",
            "PSScriptRoot": "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ansible-tmp-1465288618.21-196443209232539",
            "PipelineLength": 0,
            "PipelinePosition": 0,
            "PositionMessage": "At C:\\Users\\Administrator\\AppData\\Local\\Temp\\ansible-tmp-1465288618.21-196443209232539\\win_updates.ps1:571 char:57\r\n+   While (($job.Output -eq $null -or -not ($job.Output | Get-Member -Name Keys) - ...\r\n+~~~~~~~~~~~~~~~~~~~~~",
            "ScriptLineNumber": 571,
            "ScriptName": "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ansible-tmp-1465288618.21-196443209232539\\win_updates.ps1",
            "UnboundArguments": []
        },
        "PSMessageDetails": null,
        "PipelineIterationInfo": [],
        "ScriptStackTrace": "at RunAsScheduledJob, C:\\Users\\Administrator\\AppData\\Local\\Temp\\ansible-tmp-1465288618.21-196443209232539\\win_updates.ps1: line 571\r\nat <ScriptBlock>, C:\\Users\\Administrator\\AppData\\Local\\Temp\\ansible-tmp-1465288618.21-196443209232539\\win_updates.ps1: line 645\r\nat <ScriptBlock>, <No file>: line 4",
        "TargetObject": null
    },
    "failed": true,
    "msg": "You must specify an object for the Get-Member cmdlet."
}
```

With the task:

```
- name: Windows Update
  win_updates:
    category_names: ['SecurityUpdates', 'CriticalUpdates', 'UpdateRollups']
```

I ran the playbook again against the same instance and it worked fine - Brought up a new instance and again it ran fine so nothing reproducible as yet...

Speculating @jrquint stumbled across [this](http://stackoverflow.com/questions/23522309/must-specify-an-object-on-piping-variable-to-get-member) or similar.

**Comment by jrquint** [comment]
If I recall correctly this is exactly the issue I was having.  

--Josh

**Comment by nitzmahone** [comment]
I finally hit this yesterday on a box with Powershell 5, then spent most of the day flogging it for "WTF?!" It appears to be more evidence of the scary race conditions in Powershell scheduled jobs. If you look at the code on our end, this _shouldn't_ be possible. I've added a `-ErrorAction Ignore` to Get-Member to prevent this particular race from causing a failure, but given that it's taken me this long to hit it (and the behavior implied by this failure), I'm not sure it can't recur on the next condition (or in any other spot in the code) either. I ran a few hundred iterations of it today and didn't hit it again, but that doesn't really give me a high degree of confidence... Longer-term, I'm once again playing with different ways to escape the WinRM jail besides the task scheduler.

**Comment by gregdek** [comment]
Thanks @jrquint for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by nitzmahone** [comment]
I'm going to close this PR, since it doesn't appear make the underlying issue any better or worse.

**Pull request closed without merging**